backend/substrapp/compute_tasks/image_builder.py

```diff
@@ -24,59 +22,41 @@
-                return
+    api_function = ApiFunction.objects.get(key=function.key)
 
-            attempt += 1
-            time.sleep(IMAGE_BUILD_CHECK_DELAY)
+    attempt = 0
+    # with 60 attempts we wait max 2 min with a pending pod
+    max_attempts = IMAGE_BUILD_TIMEOUT / IMAGE_BUILD_CHECK_DELAY
+    while attempt < max_attempts:
+        if api_function.status == ApiFunction.Status.FUNCTION_STATUS_READY:
+            return
+        attempt += 1
+        time.sleep(IMAGE_BUILD_CHECK_DELAY)
+        api_function.refresh_from_db()
 
-        raise exceptions.PodTimeoutError(
-            f"Build for function {function.key} didn't complete after {IMAGE_BUILD_TIMEOUT} seconds"
-        )
-    else:
-        # Ask the backend owner of the function if it's available
-        logger.info(
-            f"Initial function URI {function.function_address.uri}; "
-            f"modified URI{function.function_address.uri.replace('file', 'image')}"
-        )
-        # TODO remove hard-coded values
-        attempt = 0
-        max_attempts = 60
-        while attempt < max_attempts:
-            try:
-                function_image_content = organization_client.get(
-                    channel=channel,
-                    organization_id=function.owner,
-                    # TODO create a clean Address for function image
-                    url=function.function_address.uri.replace("file", "image"),
-                    checksum=None,
-                )
-                break
-            except OrganizationHttpError:
-                attempt += 1
-                time.sleep(5)
-        else:
-            raise exceptions.PodTimeoutError(
-                f"Build for function {function.key} on backend {function.owner} didn't complete after 300 seconds"
-            )
+    raise exceptions.PodTimeoutError(
+        f"Build for function {function.key} didn't complete after {IMAGE_BUILD_TIMEOUT} seconds"
+    )
 
-        os.makedirs(SUBTUPLE_TMP_DIR, exist_ok=True)
-        with TemporaryDirectory(dir=SUBTUPLE_TMP_DIR) as tmp_dir:
-            storage_path = pathlib.Path(tmp_dir) / f"{container_image_tag}.zip"
-            storage_path.write_bytes(function_image_content)
-            push_payload(storage_path, registry=REGISTRY, secure=False)
 
-            #
-            # if exc.status_code == 404:
-            #     raise exceptions.CeleryRetryError(f"Function {function.key} was not found
-            #     on backend {function.owner}")
-            # else:
-            #     raise exceptions.CeleryNoRetryError("I've got a bad feeling about this.")
+def load_remote_function_image(function: orchestrator.Function, channel: str) -> None:
+    container_image_tag = utils.container_image_tag_from_function(function)
+    # Ask the backend owner of the function if it's available
+    logger.info(
+        f"Initial function URI {function.function_address.uri}; "
+        f"modified URI{function.function_address.uri.replace('file', 'image')}"
+    )
```

Comment:
Can we store the correct URI in the Function object? Maybe in this PR as we are already making changes (and a migration) to the object, or later. 
This ad-hoc replacement feels very hacky.

Reply:
This piece of code was moved and separated from backend/substrapp/compute_tasks/image_builder.py. I would be in favor of making minimal additions to this PR so we can merrge it in the POC branch, where we could start this kind of smaller changes in parallel.